src/audio/audio_stream.c

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2023 Intel Corporation. All rights reserved.
+//
+
+#include <sof/audio/audio_stream.h>
+
+static size_t audio_stream_get_free_size(struct sink __sparse_cache *sink)
+{
+	struct audio_stream __sparse_cache *audio_stream =
+			container_of(sink, struct audio_stream, sink_api);
+
+	return audio_stream_get_free_bytes(audio_stream);
+}
+
+static int audio_stream_get_buffer(struct sink __sparse_cache *sink, size_t req_size,
+				   void __sparse_cache **data_ptr,
+				   void __sparse_cache **buffer_start, size_t *buffer_size)
```

Comment:
ah, this is actually interesting now. To which memory does the attribute apply and can we control that? E.g. `data_ptr` - does the memory, to which it points now reside in the `__cache` address space, or the memory, to which `*data_ptr` points?.. It's similar to the `const` attribute: `const void **ptr` and `void * const *ptr` are two different things. But I'm not sure if moving a sparse attribute around has the same effect...